lib/matplotlib/text.py

```diff
@@ -1269,9 +1271,8 @@ def set_usetex(self, usetex):
         """
         Parameters
         ----------
-        usetex : bool or None
-            Whether to render using TeX, ``None`` means to use
-            :rc:`text.usetex`.
+        usetex : bool, default: :rc:`text.usetex`
```

Comment:
This line is both used in the [docstring of the setter](https://67257-1385122-gh.circle-artifacts.com/0/doc/build/html/api/text_api.html?highlight=text#matplotlib.text.Text.set_usetex) and in the listing of the properties (e.g. [here](https://67257-1385122-gh.circle-artifacts.com/0/doc/build/html/api/text_api.html?highlight=text#matplotlib.text.Text).

While `bool, default: ...` makes sense in the latter, it does for the setter. The setter has no default. You cannot do `set_usetex()`. To accommodate both cases, we should stick to the original `bool or None` and describe what None means verbally. That's not as nice for the listing, but at least it's correct. I'm afraid we'd need some more machinery if we wanted to be be better here.

Same holds for the other doc changes in this PR.

Reply:
I see your point. However, as the kwdoc is use in the listing of the properties, this seems to be the only way to get them there. (One way is of course to change it to a keyword-argument with default None, in that way it is slightly more correct...).

Also, see https://github.com/matplotlib/matplotlib/pull/22556#r816451833 (which I guess is why I did it).

Reply:
What I'm trying to say is that there is no good solution here. Setters and passing by keyword are conceptually different enough so that there is no wording that can correctly describe both in detail.

### Option 1 (original)

```
def set_usetex(self, usetex)
    """
    ...
    usetex : bool or None
        Whether to render using TeX, ``None`` means to use :rc:`text.usetex`.
```
**Advantage:** The setter has a reasonable API and is documented correctly. `set_usetex(None)` meaning use the default `rcParams` is a bit of a stretch but bearable (and we have precedence so it's not that easy to change).
**Disadvantage:** Property lists only show "bool or None", you have to click through to the setter to find the rcParams name and value.

### Option 2 (current PR)

```
def set_usetex(self, usetex):
    """
    ...
    usetex : bool, default: :rc:`text.usetex`
        Whether to render using TeX.
```
**Advantage:** Property lists show nicely as "bool or rcParams['text.usetex'], default: ..."
**Disadvantage:** The setter description is incorrect. usetex has no default, instead one has to pass None, which is not documented here.

### Option 3 (the above comment)

```
def set_usetex(self, usetex=None):
    """
    ...
    usetex : bool, default: :rc:`text.usetex`
        Whether to render using TeX.
```
**Advantage:** Property lists show nicely as "bool or rcParams['text.usetex'], default: ...". The setter description is correct.
**Disadvantage:** The setter API is awkward. Having `set_usetex()` mean "use the rcParams default" is quite opaque.

## Summary
Of these option 1 is the least bad. If you want to have better property listing, that must use something else than only taking the setter type description.

Reply:
Thanks for the well-made outline of the options! I do indeed see the problem. Maybe @QuLogic would like to chip in here as I interpreted his suggestion in https://github.com/matplotlib/matplotlib/pull/22556#discussion_r821544848 as he preferred Option 2.

We should probably discuss this in a dev call as it affects some other PRs as well (not to mention upcoming ones).

Reply:
If this needs some special casing, we need to update [the documentation](https://matplotlib.org/devdocs/devel/documenting_mpl.html#default-values).

Reply:
The other question would be whether such setters should accept None as "use rcparams" at all or whether defaults should be handled separately in __init__ (explicitly or using to be defined helpers).